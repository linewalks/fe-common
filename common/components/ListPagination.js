import React from 'react'

const ListPagination = ({ opts }) => {
  return (
    <select>
      {opts.map((opt, idx) => (
        <option key={`listPagination__${idx}`} value={opt.value}>
          {opt.text}
        </option>
      ))}
    </select>
  )
}
export default ListPagination
