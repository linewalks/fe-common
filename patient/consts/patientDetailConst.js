export const pVisitCategory = [
  { id: 'visitID', value: '방문 ID', tableCol: 'visitID' },
  { id: 'visitConceptName', value: '방문 유형', tableCol: 'visitConceptName' },
  { id: 'visitStartDate', value: '방문 시작일', tableCol: 'visitStartDate' },
  { id: 'visitEndDate', value: '방문 종료일', tableCol: 'visitEndDate' },
]
