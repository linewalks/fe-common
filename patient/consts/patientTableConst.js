export const initListFetch = ['race', 'gender', 'ethnicity']

export const patientCategories = [
  { id: 'personID', value: '환자 ID', tableCol: 'person_id' },
  { id: 'gender', value: '성별', tableCol: 'gender', filter: true },
  { id: 'birthDatetime', value: '생년월일', tableCol: 'birth' },
  { id: 'age', value: '나이', tableCol: 'age', filter: true },
  { id: 'race', value: '인종', tableCol: 'race', filter: true },
  { id: 'ethnicity', value: '민족', tableCol: 'ethnicity', filter: true },
  { id: 'isDeath', value: '사망여부', tableCol: 'death', filter: true },
]
export const paginationOpts = [
  { text: '10개씩 출력', value: 10 },
  { text: '20개씩 출력', value: 20 },
  { text: '50개씩 출력', value: 50 },
]

export const PAGE_CNT = 10
