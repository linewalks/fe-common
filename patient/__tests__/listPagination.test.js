import React from 'react'
import { shallow } from 'enzyme'

import ListPagination from '@common/components/ListPagination'
import { paginationOpts } from '@patient/consts/patientTableConst'

describe('환자 테이블의 <ListPagination />', () => {
  it('스냅샷 테스트', () => {
    const wrapper = shallow(
      <ListPagination opts={paginationOpts} onChange={() => {}} />,
    )
    expect(wrapper).toMatchSnapshot()
  })
})
