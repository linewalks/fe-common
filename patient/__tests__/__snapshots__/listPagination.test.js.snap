// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`환자 테이블의 <ListPagination /> 스냅샷 테스트 1`] = `
<ListPagination__WrapListPagination
  onChange={[Function]}
>
  <option
    key="listPagination__0"
    value={10}
  >
    10개씩 출력
  </option>
  <option
    key="listPagination__1"
    value={20}
  >
    20개씩 출력
  </option>
  <option
    key="listPagination__2"
    value={30}
  >
    30개씩 출력
  </option>
</ListPagination__WrapListPagination>
`;
